app/models/fannish_next_of_kin.rb

```diff
@@ -8,4 +8,26 @@ class FannishNextOfKin < ActiveRecord::Base
   def kin_name
     User.find_by(id: kin_id).try(:login)
   end
+
+  def self.update_for_user(user, kin_name, kin_email)
+    kin_user = User.find_by(login: kin_name)
+    current_fnok = user.fannish_next_of_kin
+    # first scenario: user has no existing FNOK
+    if current_fnok.nil? && kin_user.present? && kin_email.present?
+      create(
+        user_id: user.id,
+        kin_id: kin_user.id,
+        kin_email: kin_email
+      )
+    # second scenario: update user's FNOK
+    elsif kin_user.present? && kin_email.present?
+      current_fnok.update_attributes(
+        kin_id: kin_user.id,
+        kin_email: kin_email
+      )
+    # third scenario: delete user's FNOK
+    elsif kin_name.blank? && kin_email.blank?
```

Comment:
Should this check if `current_fnok.present?` before attempting to destroy it?

Reply:
I think this is the only outstanding question/last hurdle to this getting marked ready to mere. (It's possible it was answered in chat, but August was a long time ago. Sob.)

Reply:
I've pulled the code and I don't get any weird errors when the FNOK is nil, so I'll call this ready to merge.